tools/font-edit/twin-fedit.c

```diff
@@ -492,95 +497,102 @@ static void button(char_t *c, XButtonEvent *bev)
-                    spline = c->last;
-                else
-                    spline = 0;
-                if (spline) {
-                    int keysyms_keycode;
-                    KeySym *keysym = XGetKeyboardMapping(dpy, ev.xkey.keycode,
-                                                         1, &keysyms_keycode);
-                    switch (keysyms_keycode) {
-                    case XK_Left:
-                        tweak_spline(c, spline, ev.xkey.state & ShiftMask, -1,
-                                     0);
-                        draw_char(c);
-                        break;
-                    case XK_Right:
-                        tweak_spline(c, spline, ev.xkey.state & ShiftMask, 1,
-                                     0);
-                        draw_char(c);
-                        break;
-                    case XK_Up:
-                        tweak_spline(c, spline, ev.xkey.state & ShiftMask, 0,
-                                     -1);
-                        draw_char(c);
-                        break;
-                    case XK_Down:
-                        tweak_spline(c, spline, ev.xkey.state & ShiftMask, 0,
-                                     1);
-                        draw_char(c);
-                        break;
+                case SDLK_DOWN:
+                    if (spline) {
+                        int dx = 0, dy = 0;
+                        if (event.key.keysym.sym == SDLK_LEFT)
+                            dx = -1;
+                        if (event.key.keysym.sym == SDLK_RIGHT)
+                            dx = 1;
+                        if (event.key.keysym.sym == SDLK_UP)
+                            dy = -1;
+                        if (event.key.keysym.sym == SDLK_DOWN)
+                            dy = 1;
+                        tweak_spline(c, spline,
+                                     event.key.keysym.mod & KMOD_SHIFT, dx, dy);
                     }
-                    XFree(keysym);
+                    break;
                 }
+                draw_char(
+                    c);  // Ensure the content is redrawn after every key press
+                break;
+            /* End if key event detected */
+            case SDL_WINDOWEVENT:
+                if (event.window.event == SDL_WINDOWEVENT_EXPOSED)
+                    draw_char(c);
+                break;
+            case SDL_MOUSEBUTTONDOWN:
+                button(c, &event.button);
+                draw_char(
+                    c);  // Redraw the content after mouse button interaction
+                break;
+                break;
+                /* End if SDL event detected */
             }
-            break;
-        case Expose:
-            if (ev.xexpose.count == 0)
-                draw_char(c);
-            break;
-        case ButtonPress:
-            button(c, &ev.xbutton);
-            break;
         }
+
+        // Ensure the window surface is updated
```

Comment:
Always use C-style comments.

Reply:
Resolve in https://github.com/sysprog21/mado/commit/7231d62d0106fb56594ddd7785efcc9ab9779b04

SDL2 header was moved to twin-fedit.c.